api/config/config.go

```diff
@@ -61,38 +66,38 @@ type Config struct {
 
 // UIConfig stores the configuration for the UI.
 type UIConfig struct {
-	StaticPath string `envconfig:"UI_STATIC_PATH" default:"ui/build"`
-	IndexPath  string `envconfig:"UI_INDEX_PATH" default:"index.html"`
+	StaticPath string `validate:"required" default:"ui/build"`
+	IndexPath  string `validate:"required" default:"index.html"`
 }
 
 type ReactAppConfig struct {
-	DocURL            Documentations `envconfig:"REACT_APP_MERLIN_DOCS_URL" json:"REACT_APP_MERLIN_DOCS_URL,omitempty"`
-	DockerRegistries  string         `envconfig:"REACT_APP_DOCKER_REGISTRIES" json:"REACT_APP_DOCKER_REGISTRIES,omitempty"`
-	Environment       string         `envconfig:"REACT_APP_ENVIRONMENT" json:"REACT_APP_ENVIRONMENT,omitempty"`
-	FeastCoreURL      string         `envconfig:"REACT_APP_FEAST_CORE_API" json:"REACT_APP_FEAST_CORE_API,omitempty"`
-	HomePage          string         `envconfig:"REACT_APP_HOMEPAGE" json:"REACT_APP_HOMEPAGE,omitempty"`
-	MaxAllowedReplica int            `envconfig:"REACT_APP_MAX_ALLOWED_REPLICA" default:"20" json:"REACT_APP_MAX_ALLOWED_REPLICA,omitempty"`
-	MerlinURL         string         `envconfig:"REACT_APP_MERLIN_API" json:"REACT_APP_MERLIN_API,omitempty"`
-	MlpURL            string         `envconfig:"REACT_APP_MLP_API" json:"REACT_APP_MLP_API,omitempty"`
-	OauthClientID     string         `envconfig:"REACT_APP_OAUTH_CLIENT_ID" json:"REACT_APP_OAUTH_CLIENT_ID,omitempty"`
-	SentryDSN         string         `envconfig:"REACT_APP_SENTRY_DSN" json:"REACT_APP_SENTRY_DSN,omitempty"`
-	UPIDocumentation  string         `envconfig:"REACT_APP_UPI_DOC_URL" json:"REACT_APP_UPI_DOC_URL,omitempty"`
-	CPUCost           string         `envconfig:"REACT_APP_CPU_COST" json:"REACT_APP_CPU_COST,omitempty"`
-	MemoryCost        string         `envconfig:"REACT_APP_MEMORY_COST" json:"REACT_APP_MEMORY_COST,omitempty"`
+	DocURL            Documentations `json:"REACT_APP_MERLIN_DOCS_URL,omitempty"`
```

Comment:
Are we not setting the defaults here anymore?

Reply:
Hmm, there weren't default values furnished on the app-level previously 🤔 Unless I missed something... Otherwise, I can provide default values from the Helm chart values file. Would that work?

Reply:
We are not setting default value of DocURL from the API, but the UI: https://github.com/caraml-dev/merlin/blob/main/ui/src/config.js#L32-L35

So, I think it's okay to keep the status quo. No need to set default value here.